db/migrate/20191126181009_seed_test_results.rb

```diff
@@ -1,36 +1,38 @@
 class SeedTestResults < ActiveRecord::Migration[5.2]
   def up
     migrated = 0
-    test_results = []
-    rule_results_set = []
-    ProfileHost.find_each do |profile_host|
-      host = profile_host.host
-      profile = profile_host.profile
-      if host.nil? || profile.nil?
-        profile_host.destroy
-        next
+    ProfileHost.find_in_batches(batch_size: 200) do |profile_host_group|
+      test_results = []
+      rule_results_set = []
+      profile_host_group.each do |profile_host|
+        host = profile_host.host
+        profile = profile_host.profile
+        if host.nil? || profile.nil?
+          profile_host.destroy
+          next
+        end
+        puts(" - Migrating results for host #{host.name}"\
+             " - profile #{profile.name}")
+        rule_results_by_end_time = scan_results(host, profile)
+          .group_by(&:end_time)
+        rule_results_by_end_time.each do |end_time, rule_results|
+          rules_passed = rule_results.count { |rr| RuleResult::PASSED.include?(rr.result) }
+          rules_failed = rule_results.count { |rr| RuleResult::FAIL.include?(rr.result) }
+          rule_results_set << rule_results
+          test_results << {
+            host_id: host.id,
+            profile_id: profile.id,
+            start_time: rule_results[0].start_time || Time.now.utc,
+            end_time: rule_results[0].end_time || Time.now.utc,
+            score: compliance_score(rules_passed, rules_failed)
+          }
+        end
+        migrated += 1
+        imported_test_results = TestResult.import test_results
+        imported_test_results.ids.zip(rule_results_set).each do |imported_test_result, rule_results|
+          RuleResult.where(id: rule_results.pluck(:id)).update_all(test_result_id: imported_test_result)
+        end
```

Comment:
And now this runs on every batch of 200, instead of at the end of the whole thing